pulp_ansible/app/tasks/collections.py

```diff
@@ -378,12 +388,54 @@ def _build_url(path_or_url):
                         if result.get("version") and not download_url:
                             version_url = _build_url(result["href"])
                             not_done.update([remote.get_downloader(url=version_url).run()])
+                            additional_metadata[version_url] = {
+                                "docs_blob_url": f"{version_url}docs-blob/"
+                            }
 
                         if download_url:
+                            metadata = additional_metadata.get(_build_url(data["href"]), {})
+                            data["docs_blob_url"] = metadata.get("docs_blob_url")
                             yield data
                             progress_bar.increment()
 
 
+class DocsBlobDownloader(ArtifactDownloader):
+    """
+    Stage for downloading docs_blob.
+
+    Args:
+        max_concurrent_content (int): The maximum number of
+            :class:`~pulpcore.plugin.stages.DeclarativeContent` instances to handle simultaneously.
+            Default is 200.
+        args: unused positional arguments passed along to :class:`~pulpcore.plugin.stages.Stage`.
+        kwargs: unused keyword arguments passed along to :class:`~pulpcore.plugin.stages.Stage`.
+    """
+
+    async def _handle_content_unit(self, d_content):
+        """Handle one content unit.
+
+        Returns:
+            The number of downloads
+        """
+        downloaded = 0
+        if d_content.d_artifacts:
+            remote = d_content.d_artifacts[0].remote
+            docs_blob = d_content.extra_data.get("docs_blob_url")
+            if docs_blob:
+                downloaded = 1
+                downloader = remote.get_downloader(
+                    url=docs_blob, silence_errors_for_response_status_codes={404}
+                )
+                try:
+                    download_result = await downloader.run()
+                    d_content.extra_data["docs_blob_file"] = download_result.path
```

Comment:
```suggestion
                    d_content.extra_data["docs_blob_file_path"] = download_result.path
```

For clarity when it's a string path I put `_path` in the name to disambiguate versus a file handle.